cmd/boostd/backup.go

```diff
@@ -0,0 +1,274 @@
+		r, err := lotus_repo.NewFS(boostRepoPath)
+		if err != nil {
+			return err
+		}
+		ok, err := r.Exists()
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return fmt.Errorf("repo at '%s' is not initialized", cctx.String(FlagBoostRepo))
+		}
+
+		lr, err := r.LockRO(node.Boost)
+		if err != nil {
+			return fmt.Errorf("locking repo: %w", err)
+		}
+		defer lr.Close()
+
+		mds, err := lr.Datastore(context.TODO(), "/metadata")
+		if err != nil {
+			return fmt.Errorf("getting metadata datastore: %w", err)
+		}
+
+		bds, err := backupds.Wrap(mds, backupds.NoLogdir)
+		if err != nil {
+			return err
+		}
+
+		bpath, err := homedir.Expand(cctx.Path("path"))
+		if err != nil {
+			return fmt.Errorf("expanding backup directory path: %w", err)
+		}
+
+		bkp_dir := path.Join(bpath, "boost_backup_"+time.Now().Format("20060102150405"))
+
+		if err := os.Mkdir(bkp_dir, 0755); err != nil {
+			return fmt.Errorf("Error creating backup directory %s: %w", bkp_dir, err)
+		}
+
+		fpath_name := path.Join(bkp_dir, "metadata")
+
+		fpath, err := homedir.Expand(fpath_name)
+		if err != nil {
+			return fmt.Errorf("expanding metadata file path: %w", err)
+		}
+
+		out, err := os.OpenFile(fpath, os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return fmt.Errorf("opening backup file %s: %w", fpath, err)
+		}
+
+		if err := bds.Backup(cctx.Context, out); err != nil {
+			if cerr := out.Close(); cerr != nil {
+				log.Errorw("error closing backup file while handling backup error", "closeErr", cerr, "backupErr", err)
+			}
+			return fmt.Errorf("backup error: %w", err)
+		}
+
+		if err := out.Close(); err != nil {
+			return fmt.Errorf("closing backup file: %w", err)
+		}
+
+		for name, perm := range fm {
+			src_name := path.Join(lr.Path(), name)
+
+			src_path, err := homedir.Expand(src_name)
+			if err != nil {
+				return fmt.Errorf("expanding source file path %s: %w", src_name, err)
+			}
+
+			dest_name := path.Join(bkp_dir, name)
+
+			dest_path, err := homedir.Expand(dest_name)
+			if err != nil {
+				return fmt.Errorf("expanding destination file path %s: %w", dest_name, err)
+			}
+
+			if err := copy_files(src_path, dest_path, perm); err != nil {
+				return fmt.Errorf("Error copying file %s: %w", src_name, err)
+			}
+
+		}
+
+		return nil
+	},
+}
+
+var restoreCmd = &cli.Command{
+	Name:  "restore",
+	Usage: "Restores a boost repository from backup",
+	Flags: []cli.Flag{
+		&cli.PathFlag{
+			Name:     "restore-path",
+			Usage:    "Specify the path to backup directory",
+			Required: true,
+		},
+	},
+	Before: before,
+	Action: func(cctx *cli.Context) error {
+
+		repoPath := cctx.String(FlagBoostRepo)
+		fmt.Printf("Checking if repo exists at %s\n", repoPath)
+
+		r, err := lotus_repo.NewFS(repoPath)
+		if err != nil {
+			return err
+		}
+
+		ok, err := r.Exists()
+		if err != nil {
+			return err
+		}
+		if ok {
+			return fmt.Errorf("repo at '%s' is already initialized, cannot restore.", repoPath)
+		}
+
+		fmt.Println("Creating boost repo")
```

Comment:
I like these notes to the user telling them what the command is doing 👍 
Let's add the same thing to the backup command as well.